src/main/java/org/adridadou/ethereum/EthereumContractInvocationHandler.java

```diff
@@ -114,8 +114,8 @@ private Constructor lookForNonEmptyConstructor(Class<?> returnType, Object[] res
         throw new IllegalArgumentException("no constructor with arguments found! for type " + returnType.getSimpleName());
     }
 
-    <T> void register(T proxy, Class<T> contractInterface, SoliditySource code, String contractName, EthAddress address, EthAccount account) {
-        final Map<String, CompilationResult.ContractMetadata> contractsFound = compile(code.getSource()).contracts;
+    <T> void register(T proxy, Class<T> contractInterface, SoliditySource code, String contractName, EthAddress address, EthAccount account) throws IOException {
+     	final Map<String, CompilationResult.ContractMetadata> contractsFound =  code.isCompiled() ?  CompilationResult.parse(code.getRawJson()).contracts :   compile(code.getSource()).contracts;
```

Comment:
Have a look at the class CompiledContract I think it goes in the  direction of what you're trying to do.